src/ProjectTemplates/BlazorTemplates.Tests/BlazorServerTemplateTest.cs

```diff
@@ -107,84 +136,97 @@ public async Task BlazorServerTemplateWorks_IndividualAuth(bool useLocalDB)
-                if (BrowserFixture.IsHostAutomationSupported())
+                if (Fixture.BrowserManager.IsAvailable(browserKind))
                 {
-                    aspNetProcess.VisitInBrowser(Browser);
-                    TestBasicNavigation();
+                    var page = await aspNetProcess.VisitInBrowserAsync(browser);
+                    await TestBasicNavigation(page);
+                    await page.CloseAsync();
                 }
                 else
                 {
-                    BrowserFixture.EnforceSupportedConfigurations();
+                    Assert.False(
+                        TestHelpers.TryValidateBrowserRequired(
+                            browserKind,
+                            isRequired: !Fixture.BrowserManager.IsExplicitlyDisabled(browserKind),
+                            out var errorMessage),
+                        errorMessage);
                 }
             }
 
-
             using (var aspNetProcess = Project.StartPublishedProjectAsync())
             {
                 Assert.False(
                     aspNetProcess.Process.HasExited,
                     ErrorMessages.GetFailedProcessMessageOrEmpty("Run published project", Project, aspNetProcess.Process));
 
                 await aspNetProcess.AssertStatusCode("/", HttpStatusCode.OK, "text/html");
-                if (BrowserFixture.IsHostAutomationSupported())
+                if (Fixture.BrowserManager.IsAvailable(browserKind))
+                {
+                    var page = await aspNetProcess.VisitInBrowserAsync(browser);
+                    await TestBasicNavigation(page);
+                    await page.CloseAsync();
+                }
+                else
                 {
-                    aspNetProcess.VisitInBrowser(Browser);
-                    TestBasicNavigation();
+                    Assert.False(
+                        TestHelpers.TryValidateBrowserRequired(
+                            browserKind,
+                            isRequired: !Fixture.BrowserManager.IsExplicitlyDisabled(browserKind),
+                            out var errorMessage),
+                        errorMessage);
                 }
             }
         }
 
-        private void TestBasicNavigation()
+        private async Task TestBasicNavigation(IPage page)
         {
-            var retries = 3;
-            var connected = false;
-            do
-            {
-                try
-                {
-                    Browser.Contains("Information: WebSocket connected to",
-                        () => string.Join(Environment.NewLine, Browser.GetBrowserLogs(LogLevel.Info).Select(b => b.Message)));
-                    connected = true;
-                }
-                catch (TimeoutException) when(retries-- > 0)
-                {
-                    Browser.Navigate().Refresh();
-                }
-            } while (!connected && retries > 0);
```

Comment:
I can't really tell what purpose this logic served before. Maybe it's fine to remove it. Do you know anything about this?

Reply:
This was to detect the websocket had started, but I'm replacing it with different logic that actually detects when the websocket has been established and when the first render batch has been received (events are wired up) before continuing.